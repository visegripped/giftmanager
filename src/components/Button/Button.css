.button {
  background: rgb(254, 127, 0);
  color: #fff;
  border: 0;
  border-radius: 8px;
  padding: 4px 12px;
  font-size: 1rem;
}