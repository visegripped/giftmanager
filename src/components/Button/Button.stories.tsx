import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';

import Button from './index';
import { ButtonProps } from './Button';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'component/Button',
  component: Button,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
} as ComponentMeta<typeof Button>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof Button> = (args) => <Button {...args} />;

const getButtonsByPriority = (priority: ButtonProps['priority']) => {
  const themes = ['standard', 'info', 'warn', 'error', ''];
  return (
    <>
      {themes.map((theme) => {
        return (
          <Button priority={priority} theme={theme} onClick={() => null}>
            Secondary {theme}
          </Button>
        );
      })}
    </>
  );
};

export const Primary = Template.bind({});
// More on args: https://storybook.js.org/docs/react/writing-stories/args
Primary.decorators = [
  () => {
    return getButtonsByPriority('primary');
  },
];

export const Secondary = Template.bind({});
Secondary.decorators = [
  () => {
    return getButtonsByPriority('secondary');
  },
];
export const Ghost = Template.bind({});
Ghost.decorators = [
  () => {
    return getButtonsByPriority('ghost');
  },
];

export const Disabled = Template.bind({});
Disabled.args = {
  children: <>Disabled</>,
  priority: 'primary',
  disabled: true,
};
