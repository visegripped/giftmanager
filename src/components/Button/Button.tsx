import React from 'react';
import './Button.css';

interface ButtonProps {
  text: string; // change this to use children
  //   children: React.ReactNode;
  cssClasses?: string;
  disabled?: boolean;
  priority?: 'primary' | 'secondary' | 'ghost' | 'error' | 'critical' | 'info'; // error shouldn't be a priority.  Add a new theme prop w/ standard, error, warn
  onClick(clickEvent: object, props: object): void;
  id?: string;
}

const Button = (props: ButtonProps) => {
  const { text, cssClasses, id, disabled, priority, onClick = () => null } = props;
  const buttonClickHandler = (clickEvent: React.MouseEvent<HTMLButtonElement>) => {
    onClick(clickEvent, props);
  };
  const className = `button ${priority} ${cssClasses}`;
  return (
    <button id={id} className={className} disabled={disabled} data-testid="Button" onClick={buttonClickHandler}>
      {text}
    </button>
  );
};

export default Button;
