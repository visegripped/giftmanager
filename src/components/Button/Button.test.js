import { render, fireEvent, screen } from "@testing-library/react";
import Button from "./index";

describe("Button", () => {
  test("onClick is executed properly", () => {
    let counter = 0;
    const testForClick = () => {
      counter += 1;
    };
    render(<Button onClick={testForClick} />);
    const renderedButton = screen.getByTestId("button");
    fireEvent.click(renderedButton);
    expect(counter).toEqual(1);
  });
  test("snapshot matches", () => {
    render(
      <Button
        text="click me"
        cssClasses="test classes"
        priority="secondary"
        disabled={false}
      />
    );
    const renderedButton = screen.getByTestId("button");
    expect(renderedButton).toMatchSnapshot();
  });
});
