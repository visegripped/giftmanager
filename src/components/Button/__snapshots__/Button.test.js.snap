// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot matches 1`] = `
<button
  class="button secondary test classes"
  data-testid="button"
>
  click me
</button>
`;
